solution/index.js

```diff
@@ -0,0 +1,356 @@
+    if (category === 'to-do') category = 'todo'
+    parsedTasks[category].splice(
+      parsedTasks[category].indexOf(originalTextContent),
+      1,
+      blurEvent.target.textContent
+    )
+    localStorage.setItem('tasks', JSON.stringify(parsedTasks))
+    e.target.setAttribute('contenteditable', 'false')
+  }
+}
+
+document.querySelector('#search').addEventListener('input', globalSearch)
+
+function globalSearch(e) {
+  const liList = document.querySelectorAll('li')
+  liList.forEach((li) => {
+    if (
+      !li.textContent
+        .toLocaleLowerCase()
+        .includes(e.target.value.toLocaleLowerCase())
+    ) {
+      li.className = 'hidden'
+    } else {
+      li.className = 'task list-item'
+    }
+  })
+}
+
+function createLoader() {
+  const container = document.querySelector('#container')
+  const loader = document.createElement('img')
+  loader.className = 'loader'
+  loader.src = '../gifs/loader.gif'
+  loader.alt = 'image of loader'
+  document.body.insertBefore(loader, container)
+  return loader
+}
+
+document.querySelector('#load-btn').addEventListener('click', loadFromApi)
+document.querySelector('#save-btn').addEventListener('click', saveToApi)
+
+async function loadFromApi() {
+  try {
+    const loader = createLoader()
+    const url = 'https://json-bins.herokuapp.com/bin/614afda34021ac0e6c080cc9'
+    const response = await fetch(url, {
+      method: 'Get',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+    })
+    // handle errors:
+    if (!response.ok) {
+      alert(response.status)
+      throw `HTTP Error ${response.status}`
+    }
+    const data = await response.json()
+    if (localStorage.tasks === data.tasks) {
+      return loader.remove()
+    } else {
+      const newString = makeNormalString(JSON.stringify(data.tasks)) // stringifies without `\\`
+      localStorage.setItem('tasks', newString)
+    }
+    loader.remove()
+    location.reload()
+  } catch (e) {
+    console.log(e)
+  }
+}
+
+async function saveToApi() {
+  try {
+    const { tasks } = localStorage
+    const loader = createLoader()
+    const url = 'https://json-bins.herokuapp.com/bin/614afda34021ac0e6c080cc9'
+    const response = await fetch(url, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ tasks }),
+    })
+    if (!response.ok) {
+      alert(response.status)
+      throw `HTTP ERROR ${response.status}`
+    }
+    loader.remove()
+  } catch (e) {
+    console.log(e)
+  }
+}
+
+function makeNormalString(tasks) {
+  tasks = tasks.replace('"{', '{')
+  tasks = tasks.replace('}"', '}')
+  return tasks.replaceAll('\\', '')
+}
+
+liS.forEach((li) => {
+  li.addEventListener('dragstart', () => {
+    li.classList.add('dragging')
+  })
+})
+
+liS.forEach((li) => {
+  li.addEventListener('dragend', () => {
+    li.classList.remove('dragging')
+  })
+})
+
+uls.forEach((ul) => {
```

Comment:
The software architecture needs to be renovated. Functions are used to solve problems that should not exist in the first place

Reply:
Thanks Ofer :)
I assume you're talking about the json?
If so I committed a better solution in another branch https://github.com/gabbyMill/kanban-final/blob/JSON-mixup-fix/solution/index.js

If not, I'd love to hear specifically which part you meant, or is it the whole of the code.
Appreciate the input, thanks :)